lib/main.js

```diff
@@ -131,13 +131,15 @@ Lambda.prototype._params = function (program, buffer) {
       Variables: config
     }
   }
-  if (program.deadLetterConfigTargetArn !== undefined) {
+  if (program.deadLetterConfigTargetArn !== undefined && program.deadLetterConfigTargetArn !== '' && program.deadLetterConfigTargetArn !== null) {
```

Comment:
as @abetomo already mentioned, we allowed `''` to be passed, to have a possibility to unset it. We would need some kind of mechanism for understanding if this is an update or a create, and adjust accordingly.